### DISCUSSION TITLE
How to trigger event using livewire edit form

### DISCUSSION THREAD

**Discussion by stojankukrika** [livewire/livewire#5554] [1. Help]
I would like to have two use cases in the same group of liveware components. For example, if the member does not select and when comes from a member and wants to be that member selected on the list for example to enter payments.
Here is my mount method:

    public function mount($member = null)
    {
        $this->payment = [
            'member_id' => $this->member,
            'amount' => 0,
            'payment_method' => null,
            'date' => null,
            'name' => null,
            'payer_name' => null,
        ];
        $this->member = $member;
    }
    public function render()
    {
        $this->members =....
    
        return view('livewire.create-team-member-membership');
    }

Select binder works OK, select right member:

    <select class="block w-full mt-1 text-sm dark:text-gray-800 dark:border-gray-600 dark:bg-white
                    form-select focus:outline-none focus:shadow-outline-purple dark:focus:shadow-outline-gray"
                    name="member_id" id="member_id" wire:model="payment.member_id" wire:change="changedMember">
                <option value="">{{__('site.member_payments.select_member')}}</option>
                @foreach($members as $item)
                    <option value="{{$item->id}}">{{$item->name}}</option>
                @endforeach
            </select>

but for some reason, changedMember didn't run well. That method emit event which shows other data for selected member:

`public function changedMember()
{
$this->emit('memberChanged', ['member_id' => $this->payment['member_id']]);
}`

but it fired when I manually select some other member. Is there any way I can call it for that already selected member?

**Comment by joshhanley** [comment]
@stojankukrika I'm sorry I don't follow what you're asking. Can you explain it in more detail? Also you're referencing `changedMember()` method but i don't see it there, can you make sure to include all relevant code. Thanks!

**Reply by stojankukrika** [reply]
I am trying to make some kind an edit form. When I open the payment form it loads all members from the database and when I click on some members it loads additional fields because the trigger changedMember() method emits an event that other components listen and all works well. But in the case when I want to add payment from member details, I like that member will be preselected and that works well with submitting member id on request, and liveware forms,.... but it didn't trigger changeMember event to other components know that needs to load the rest of data that they load on members select changed in the first scenario(need to change the value in a list of members to that would be triggered).

**Reply by joshhanley** [reply]
@stojankukrika ok, I think I follow. I'm going to need to see all of the relevant code though and what steps you are following (click button, select member, etc), so I can then try and see why the events are not firing.
